storage/buffer.go

```diff
@@ -77,13 +81,20 @@ func (b *BufferedSeriesIterator) Seek(t int64) bool {
 
 // Next advances the iterator to the next element.
 func (b *BufferedSeriesIterator) Next() bool {
+	if b.done {
+		return false
+	}
+
 	// Add current element to buffer before advancing.
 	b.buf.add(b.it.At())
 
 	ok := b.it.Next()
 	if ok {
 		b.lastTime, _ = b.Values()
+	} else {
+		b.done = true
 	}
+
```

Comment:
Would it be easier to grasp if we made it `b.ok` instead of `b.done` and have calls to `b.it.{Seek,Next}` assign to `b.ok` directly?